src/Akka.Streams.Kafka.Tests/Internal/CommitCollectorStageSpecs.cs

```diff
@@ -486,7 +487,9 @@ public override Task CommitSingle(TopicPartition topicPartition, OffsetAndMetada
 
         public override Task CommitOneOfMany(TopicPartition topicPartition, OffsetAndMetadata offsetAndMetadata)
         {
-            var commitOffset = offsetAndMetadata.Offset;
+            // CommittableOffsetBatch.OffsetsAndMetadata points the next committed message.
+            // So to get committed message offset we need to subtract 1
+            var commitOffset = offsetAndMetadata.Offset - 1;
```

Comment:
Verifies the fix inside the `TestCommitter` - JVM has these lines too (we did not, previously)